File: SakhaTyla.Core/Requests/BookAuthorships/Models/BookAuthorshipModel.cs
﻿using System;
using System.Collections.Generic;
using System.ComponentModel;

namespace SakhaTyla.Core.Requests.BookAuthorships.Models
{
    public class BookAuthorshipModel
    {
        public BookAuthorshipModel()
        {
        }

        [DisplayName("Id")]
        public int Id { get; set; }

        [DisplayName("Creation Date")]
        public DateTime CreationDate { get; set; }

        [DisplayName("Modification Date")]
        public DateTime ModificationDate { get; set; }

        [DisplayName("Creation User")]
        public int? CreationUserId { get; set; }
        public Users.Models.UserShortModel? CreationUser { get; set; }

        [DisplayName("Modification User")]
        public int? ModificationUserId { get; set; }
        public Users.Models.UserShortModel? ModificationUser { get; set; }

        [DisplayName("Book")]
        public int BookId { get; set; }
        public Books.Models.BookShortModel Book { get; set; } = null!;

        [DisplayName("Author")]
        public int AuthorId { get; set; }
        public BookAuthors.Models.BookAuthorShortModel Author { get; set; } = null!;

        [DisplayName("Weight")]
        public int Weight { get; set; }
    }
}

File: SakhaTyla.Web.Ng/src/app/book-authorship-core/book-authorship.model.ts
import { OrderDirection } from '../core/models/order-direction.model';
import { PageSettings } from '../core/page.model';
import { BookAuthorshipFilter } from './book-authorship-filter.model';

export class BookAuthorship {
  id: number;
  creationDate: Date;
  modificationDate: Date;
  creationUserId: number;
  modificationUserId: number;
  bookId: number;
  authorId: number;
  weight: number;
}

export class BookAuthorshipListState {
  pageSize = PageSettings.pageSize;
  pageIndex = 0;
  filter = new BookAuthorshipFilter();
  orderBy?: string;
  orderDirection?: OrderDirection;
}
